kbase-extension/static/kbase/js/widgets/function_output/kbaseAnnotatedMetagenomeAssembly.js

```diff
@@ -1056,126 +1066,177 @@ define ([
+                        }
+
+                        var liElems = $tabPane.find('li');
 
-                var liElems = $tabPane.find('li');
-                for (var liElemPos = 0; liElemPos < liElems.length; liElemPos++) {
-                    var liElem = $(liElems.get(liElemPos));
-                    var aElem = liElem.find('a');
-                    if (aElem.length != 1)
-                        continue;
-                    var dataTab = aElem.attr('data-tab');
-                    var metagenome_ref = self.metagenome_ref;
-                    if (dataTab === 'Browse Features' ) {
-                        aElem.on('click', function() {
+                        var browse_features_func = function (metagenome_ref) {
                             self.buildGeneSearchView({
-                                $div: $('#'+pref+'browse_features'),
+                                $div: $('#' + pref + 'browse_features'),
                                 ref: metagenome_ref,
-                                idClick: function(featureData) {
+                                idClick: function (featureData) {
                                     self.showFeatureTab(metagenome_ref, featureData, pref, tabObj);
                                 },
-                                contigClick: function(contigId) {
+                                contigClick: function (contigId) {
                                     self.showContigTab(metagenome_ref, contigId, pref, tabObj);
                                 }
                             });
-                        });
-                    } else if (dataTab === 'Browse Contigs' ) {
-                        aElem.on('click', function() {
+                        };
+
+                        var browse_contig_func = function (metagenome_ref) {
                             self.buildContigSearchView({
-                                $div: $('#'+pref+'browse_contigs'),
+                                $div: $('#' + pref + 'browse_contigs'),
                                 ref: metagenome_ref,
-                                contigClick: function(contigId) {
+                                contigClick: function (contigId) {
                                     self.showContigTab(metagenome_ref, contigId, pref, tabObj);
                                 }
                             });
-                        });
-                    }
-                }
 
+                        };
+
+                        for (let liElemPos = 0; liElemPos < liElems.length; liElemPos += 1) {
+                            var liElem = $(liElems.get(liElemPos));
+                            var aElem = liElem.find('a');
+                            if (aElem.length == 1) {
+                                var dataTab = aElem.attr('data-tab');
+                                if (dataTab === 'Browse Features' ) {
+                                    aElem.on('click', browse_features_func(metagenome_ref));
```

Comment:
For these two `aElem.on('click')` functions, they should take a function as an argument. You're executing a function inline here, which'll get run as soon as the `aElem.on` statement gets run, NOT when the click happens as you expect.

In this case, you can pass an anonymous function. I would use an arrow function, like this:
```
aElem.on('click', () => browse_features_func(metagenome_ref));
```